### PR TITLE
HSEARCH-5038 Align vector similarity naming between backends / HSEARCH-5056 Rename VectorField#beamWidth / VectorField#maxConnections to align with Elasticsearch/OpenSearch naming

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-5038
https://hibernate.atlassian.net/browse/HSEARCH-5056

@fax4ever this particular change (the last commit) may be interesting for you since it'll rename one constant (`INNER_PRODUCT` -> `DOT_PRODUCT`) and add one more (`MAX_INNER_PRODUCT`) in the `VectorSimilarity` enum.

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java [hunk 1]
```diff
@@ -883,4 +884,9 @@ SearchException vectorKnnMatchVectorTypeDiffersFromField(String absoluteFieldPat
 			value = "The Elastic distribution of Elasticsearch does not allow to apply constant score to a knn predicate."
 					+ " Constant score will not be applied.")
 	void elasticsearchKnnIgnoresConstantScore();
+
+	@Message(id = ID_OFFSET + 187,
+			value = "The OpenSearch distribution does not allow to use %1$s as a space type for a Lucene engine."
```
Comment (yrodiere) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/Log.java:889]:
```suggestion
			value = "The OpenSearch distribution does not allow using %1$s as a space type for a Lucene engine."
```

**Review by yrodiere — APPROVED** [review]
LGTM, thanks!

**Comment by yrodiere** [comment]
@marko-bekhta Looks like this needs rebasing :D

**Comment by marko-bekhta** [comment]
😱 😨 😅

**Pull request merged**